sdks/python/apache_beam/examples/snippets/snippets.py

```diff
@@ -1170,3 +1170,47 @@ def expand(self, pcoll):
         | 'PairWithOne' >> beam.Map(lambda v: (v, 1))
         | beam.CombinePerKey(sum))
 # [END model_library_transforms_count]
+
+def examples_troubleshooting_metrics(renames):
+  """DebuggingWordCount example snippets."""
+  import re
+
+  import apache_beam as beam
+
+  # [START metrics_usage_example]
+  class FilterTextFn(beam.DoFn):
+    """A DoFn that filters for a specific key based on a regular expression."""
+
+    def __init__(self, pattern):
+      self.pattern = pattern
+      # A custom metric can track values in your pipeline as it runs. Create
+      # custom metrics matched_words and unmatched_words.
+      self.matched_words = Metrics.counter(self.__class__, 'matched_words')
+      self.umatched_words = Metrics.counter(self.__class__, 'umatched_words')
+
+    def process(self, element):
+      word, _ = element
+      if re.match(self.pattern, word):
+        # Add 1 to the custom metric counter matched_words
+        self.matched_words.inc()
+        yield element
+      else:
+        # Add 1 to the custom metric counter umatched_words
+        self.umatched_words.inc()
+
+  p = TestPipeline()  # Use TestPipeline for testing.
+  filtered_words = (
+      p
+      | beam.io.ReadFromText(
```

Comment:
Don't read from a file unless you're trying to demonstrate file reads. Just start with Create(...) (and outside of the example).